### PR TITLE
Docs for the `pydantic` migration guide

### PR DESCRIPTION
**Pull request by bcdurak**
## Describe changes
I have written a docs page regarding the changes coming with the pydantic branch.

## Pre-requisites
Please ensure you have done the following:
- [X] I have read the **CONTRIBUTING.md** document.
- [X] If my change requires a change to docs, I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [X] I have based my new branch on `develop` and the open PR is targeting `develop`. If your branch wasn't based on develop read [Contribution guide on rebasing branch to develop](https://github.com/zenml-io/zenml/blob/main/CONTRIBUTING.md#-pull-requests-rebase-your-branch-on-develop).
- [ ] If my changes require changes to the dashboard, these changes are communicated/requested.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [X] Other (add details above)

### CODE REVIEW

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 1]
```diff
@@ -0,0 +1,160 @@
+---
+description: How to migrate from ZenML 0.58.2 to 0.60.0.
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:2]:
```suggestion
description: How to migrate from ZenML 0.58.2 to 0.60.0 (Pydantic 2 edition).
```

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+---
+
+# Release Notes
+
+ZenML now uses Pydantic V2. 🥳
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:7]:
```suggestion
ZenML now uses Pydantic v2. 🥳
```

They use the lowercase.

Comment (strickvl) [reply docs/book/reference/migration-guide/migration-zero-sixty.md:7]: And change it everywhere else below?

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+
+### Airflow
+
+As mentioned above upgrading our `pydantic` dependency meant we had to upgrade
+our `sqlmodel` dependency. Upgrading our `sqlmodel` dependency meant we had to
+upgrade our `sqlalchemy` dependency. Unfortunately, `apache-airflow` is still
+using `sqlalchemy` v1 and is incompatible with v2. As a solution, we have
+removed the dependencies of the `airflow` integration. Now, you can use ZenML to
+create your Airflow pipelines and use a separate environment to run them with
+Airflow.
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:58]: Do we need to link to the docs where we explain how to do this?

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+### Feast
+
+Our previous implementation of the `feast` integration was not compatible with
+Pydantic V2 due to the extra `redis` dependency we were using. This extra
+dependency is now removed and the `feast` integration is working as intended.
+
+### GCP
+
+The previous version of the kubeflow dependency (`kfp==1.8.22`) in our GCP
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:83]:
```suggestion
The previous version of the Kubeflow dependency (`kfp==1.8.22`) in our GCP
```

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 5]
```diff
@@ -0,0 +1,160 @@
...
+dependency (`kfp==1.8.22`) in our `kubeflow` integration required Pydantic V1 to
+be installed. While we were upgrading our Pydantic dependency, we saw this as an
+opportunity and wanted to use this chance to upgrade the `kfp` dependency to
+V2 (which has no dependencies on the Pydantic library). If you would like to go
+through the changes in the `kfp` library, you can
+find [the migration guide here](https://www.kubeflow.org/docs/components/pipelines/v2/migration/). (
+We also are considering adding an alternative version of this integration so our
+users can keep using `kfp` V1 in their environment. Stay tuned for any updates.)
+
+### mlflow
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:112]:
```suggestion
### MLflow
```

File: docs/book/reference/migration-guide/migration-zero-sixty.md [hunk 6]
```diff
@@ -0,0 +1,160 @@
...
+using some features of `pydantic` V1 which are deprecated. So, if the
+integration is installed in your environment, you might run into some
+deprecation warnings.
+
+### Label Studio
+
+While we were working on updating our `pydantic` dependency,
+the `label-studio-sdk` has released its 1.0 version. In this new
+version, `pydantic` V2 is also supported. The implementation and documentation
+of our label-studio integration have been updated accordingly.
```
Comment (strickvl) [inline docs/book/reference/migration-guide/migration-zero-sixty.md:128]:
```suggestion
of our Label Studio integration have been updated accordingly.
```

**Review by strickvl — CHANGES_REQUESTED** [review]
The only thing I'd maybe consider adding is more of a highlight of the ways KFP v2 isn't feature-complete yet. (@safoinme knows those details, but at the moment I think it paints a picture of more or less feature parity whereas I think there are still some things that users might want that aren't implemented by the `kfp` team.) Otherwise this looks good, minus the small nits below...

File: docs/book/reference/migration-guide/migration-zero-sixty.md (hunk 2 above)
Comment (bcdurak) [reply docs/book/reference/migration-guide/migration-zero-sixty.md]: replaced all.

File: docs/book/reference/migration-guide/migration-zero-sixty.md (hunk 3 above)
Comment (bcdurak) [reply docs/book/reference/migration-guide/migration-zero-sixty.md]: I added a link to the docs of the Airflow 👍

**Comment by bcdurak** [comment]
@strickvl Since you mentioned the Airflow docs, I added an extra paragraph down below, in case people end up with some dependency issues after they upgrade. 

also for the kfp explanation, I will contact @safoinme

**Comment by strickvl** [comment]
Ok I'll wait for that KFP update and then review

**Review by strickvl — APPROVED** [review]

**Pull request merged**